SimulationAgent/Simulation/DeviceStatusLogic/Connect.cs

```diff
@@ -69,41 +113,37 @@ public void Run(object context)
 
                 try
                 {
-                    lock (actor)
+                    var device = await this.devices.GetOrCreateAsync(this.deviceId, false, actor.CancellationToken);
```

Comment:
why remove locking the actor?  doesn't seem like you needed to lock it as there's only one thread running against it right here but thought I'd check.

Reply:
you suggested to remove the lock in the previous review :-), I agree it's not needed